src/EventListener/ProductPublisher.php

```diff
@@ -38,8 +60,27 @@ public function onFlush(OnFlushEventArgs $event)
         $scheduledInsertions = $event->getEntityManager()->getUnitOfWork()->getScheduledEntityInsertions();
 
         foreach ($scheduledInsertions as $entity) {
-            if ($entity instanceof ProductInterface) {
-                $this->scheduledProducts[] = $entity;
+            $entity = $this->getProductFromEntity($entity);
+            if ($entity instanceof ProductInterface && !isset($this->scheduledProducts[$entity->getId()])) {
+                $this->scheduledProducts[$entity->getId()] = $entity;
```

Comment:
At this point I'm not sure whether entity has an ID, we should probably use code here.

Reply:
Need to confirm if that's the case, but code does work too.